src/libmongoc/tests/test-mongoc-client-pool.c

```diff
@@ -361,15 +361,43 @@ test_client_pool_destroy_without_pushing (void)
    bson_destroy (cmd);
 }
 
+/* tests that creating and destroying session is ok */
+static void
+test_client_pool_create_unused_session (void)
+{
+   mongoc_client_t *client;
+   mongoc_client_pool_t *pool;
+   mongoc_uri_t *uri;
+   mongoc_client_session_t *session;
+
+   bson_error_t error;
+
+   capture_logs (true);
+
+   uri = mongoc_uri_new_with_error ("mongodb://127.0.0.1/", &error);
+   ASSERT_NO_CAPTURED_LOGS (error.message);
+
+   pool = mongoc_client_pool_new (uri);
```

Comment:
In addition to checking that nothing was logged, I suggest also asserting that "endSessions" was never sent.

Other tests in `test-mongoc-client-session.c` make similar assertions by setting APM callbacks in `endsessions_test_init`.

There is some background on APM events in the documentation here: http://mongoc.org/libmongoc/current/application-performance-monitoring.html

For this test, callbacks can be set on the pool, like:

```
mongoc_apm_callbacks_t *callbacks;
int count = 0;

callbacks = mongoc_apm_callbacks_new ();
mongoc_apm_set_command_started_cb (callbacks, command_started_cb);
mongoc_client_pool_set_apm_callbacks (pool, callbacks, &count);
```

Then, define `command_started_cb` like so:
```
static void
command_started_cb (const mongoc_apm_command_started_t *event)
{
   int *count;

   if (strcmp (mongoc_apm_command_started_get_command_name (event),
               "endSessions") != 0) {
      return;
   }

   count = (int *) mongoc_apm_command_started_get_context (event);
   count++;
}
```

`count` is passed through the `void *` context. After the test calls `mongoc_client_pool_destroy`, assert that `count` is still 0 with `ASSERT_CMPINT(count, ==, 0)`.